dbt_platform_helper/utils/validation.py

```diff
@@ -569,13 +569,16 @@ def _validate_extension_supported_versions(
 
         if not isinstance(environments, dict):
             click.secho(
-                "Error: Opensearch extension definition is invalid type, expected dictionary",
+                f"Error: {extension_type} extension definition is invalid type, expected dictionary",
                 fg="red",
             )
             continue
         for environment, env_config in environments.items():
+
+            # An extension version doesn't need to be specified for all environments, provided one is specified under "*".
+            # So check if the version is set before checking if it's supported
             extension_version = env_config.get(version_key)
-            if extension_version not in supported_extension_versions:
+            if extension_version and extension_version not in supported_extension_versions:
```

Comment:
I think this might let "no version" slip through.

Can we add test to verify the correction validation behaviour occurs when:

* there is no version for the default or specific environment
* there is no version for the default, but it is specified for the specific environment

Then perhaps the code here needs to work out what the version is for each environment, falling back to the default (`"*"`) as required, before validating it.

Reply:
I thought about adding code to check that a engine is specified for each environment, or there is an engine defined at the * level. 

That sort of validation should be delegated to the platform config schema no? This method is looking only that if the version is specified that it falls within supported versions.

Will add a test for above - just to clarify:

- there is no version for the default or specific environment
Would cause no warnings to be raised as no version specified
- there is no version for the default, but it is specified for the specific environment
Would cause no warning to be raised unless the version set for a specific environment is outside supported versions.

Reply:
Cool.